### PR TITLE
[BUG] Fix incorrect spark metrics

### PR DESCRIPTION
**Pull request by zuston**
### What changes were proposed in this pull request?

Fix incorrect spark metrics

### Why are the changes needed?

1. The corresponding shuffle-read records number and shuffle-write records number is not consistent in our internal cluster
2. Log wont show the correct fetch bytes, always return 0 like 

`22/11/15 13:54:53 INFO RssShuffleDataIterator: Fetch 0 bytes cost 30791 ms and 53 ms to serialize, 347 ms to decompress with unCompressionLength[274815736]
`

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

1. UTs
2. Online spark3 jobs test

### CODE REVIEW

**Comment by zuston** [comment]
PTAL @jerqi @kaijchen @leixm

**Comment by frankliee** [comment]
Good job !

Could you fix the same issue of MR client, it also use uncompressed length.

File: integration-test/spark-common/src/test/java/org/apache/uniffle/test/WriteAndReadMetricsTest.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+    stageData =
+        scala.collection.JavaConverters.seqAsJavaList(spark.sparkContext().statusStore().stageData(1, false));
+    long readRecords = stageData.get(0).shuffleReadRecords();
+
+    assertEquals(writeRecords, readRecords);
+
+    return result;
+  }
+}
```
Comment (jerqi) [inline integration-test/spark-common/src/test/java/org/apache/uniffle/test/WriteAndReadMetricsTest.java:66]: Add a new line.

Comment (zuston) [reply integration-test/spark-common/src/test/java/org/apache/uniffle/test/WriteAndReadMetricsTest.java:66]: Got it

Comment (zuston) [reply integration-test/spark-common/src/test/java/org/apache/uniffle/test/WriteAndReadMetricsTest.java]: Done

**Comment by zuston** [comment]
I'm confused that the `WriteAndReadMetricsTest` will fail in spark3.2.x . Could you help check why? @jerqi

**Comment by jerqi** [comment]
Could you debug in your local machine?

**Comment by zuston** [comment]
> Could you debug in your local machine?

Fixed @jerqi PTAL

**Review by jerqi — APPROVED** [review]
LGTM, thanks @zuston

File: integration-test/spark-common/src/test/java/org/apache/uniffle/test/WriteAndReadMetricsTest.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.functions;
+import org.apache.spark.status.AppStatusStore;
+import org.apache.spark.status.api.v1.StageData;
+import org.junit.jupiter.api.Test;
+import scala.collection.Seq;
+
+public class WriteAndReadMetricsTest extends SimpleTestBase {
```
Comment (jerqi) [inline integration-test/spark-common/src/test/java/org/apache/uniffle/test/WriteAndReadMetricsTest.java:35]: The test case seems flaky
https://github.com/apache/incubator-uniffle/actions/runs/3511234782/jobs/5881817090

**Pull request merged**